src/App/component/AddRepositoryDialog.jsx

```diff
@@ -7,63 +7,261 @@ import {
   DialogContentText,
   DialogTitle,
-  TextField
+  TextField,
+  Radio,
+  RadioGroup,
+  FormControl,
+  FormLabel,
+  FormControlLabel
 } from '@material-ui/core'
 
-export default function AddRepositoryDialog({open, reloadProjects, handleClose, projectId, repoType}) {
+import InputAdornment from '@material-ui/core/InputAdornment';
+import {SiGithub, SiSonarqube, SiGitlab} from 'react-icons/si'
+
+export default function AddRepositoryDialog({open, reloadProjects, handleClose, projectId}) {
   const [repositoryURL, setRepositoryURL] = useState("")
+  const [repoType, setRepoType] = useState("")
+  const [githubRepositoryURL, setGithubRepositoryURL] = useState("")
+  const [gitlabRepositoryURL, setGitlabRepositoryURL] = useState("")
+  const [sonarRepositoryURL, setSonarRepositoryURL] = useState("")
+  const setIsGithubAvailable = useState(false)[1]
+  const setIsGitlabAvailable = useState(false)[1]
+  const setIsSonarAvailable = useState(false)[1]
   const jwtToken = localStorage.getItem("jwtToken")
 
+  const [showGithubDiv, setShowGithubDiv] = useState(false)
+  const [showGitlabDiv, setShowGitlabDiv] = useState(false)
+  const [showSonarDiv, setShowSonarDiv] = useState(false)
+
   const addRepository = () => {
-    if (repositoryURL === "") {
+    let checker = []
+    if (githubRepositoryURL === "" && sonarRepositoryURL === "" && gitlabRepositoryURL === "") {
       alert("不準啦馬的>///<")
     } else {
-      let payload = {
-        projectId: projectId,
-        repositoryURL: repositoryURL
+
+      if (githubRepositoryURL !== "") {
+        checker.push(checkGithubRepositoryURL());
+      }
+      if (gitlabRepositoryURL !== "") {
+        checker.push(checkGitlabRepositoryURL());
       }
-      Axios.post(`http://localhost:9100/pvs-api/project/${projectId}/repository/${repoType}`, payload,
-        {headers: {"Authorization": `${jwtToken}`}})
-        .then(() => {
-          reloadProjects()
-          handleClose()
-        })
-        .catch((e) => {
-          alert(e.response.status)
-          console.error(e)
-        })
+      if (sonarRepositoryURL !== "") {
+        checker.push(checkSonarRepositoryURL());
+      }
+
+      Promise.all(checker)
+        .then((response) => {
+          if (response.includes(false) === false) {
+            let payload = {
+              projectId: projectId,
+              repositoryURL: repositoryURL
+            }
+
+            Axios.post(`http://localhost:9100/pvs-api/project/${projectId}/repository/${repoType}`, payload,
+              {headers: {"Authorization": `${jwtToken}`}})
+              .then(() => {
+                reloadProjects()
+                handleClose()
+              })
+              .catch((e) => {
+                alert(e.response.status)
+                console.error(e)
+              })
+          }
+        }).catch((e) => {
+        alert(e.response.status)
+        console.error(e)
+      })
     }
   }
 
+  const checkGithubRepositoryURL = () => {
```

Comment:
Curious about those implementations, (`check....URL`).
Although the backend should keep the `zero-trust` principle to read requests from client, but the client-side doesn't need the backend to complete its verification, we don't need to use backend to do this.
So suggest let the front-end do these checks it self.